site/en/docs/extensions/reference/storage/index.md

```diff
@@ -81,50 +82,115 @@ tubes can be filled, and if they are filled when you put your message in, it get
+If you're interested in tracking changes made to a data object, you can add a listener to its
+`onChanged` event. Whenever anything changes in storage, that event fires. Here's sample code to
+listen for saved changes:
 
 ```js
-function saveChanges() {
-  // Get a value saved in a form.
-  var theValue = textarea.value;
-  // Check that there's some code there.
-  if (!theValue) {
-    message('Error: No value specified');
-    return;
+//// background.js ////
+
+chrome.storage.onChanged.addListener(function (changes, namespace) {
+  for (let [key, { oldValue, newValue }] of Object.entries(changes)) {
+    console.log(
+      `Storage key "${key}" in namespace "${namespace}" changed.`,
+      `Old value was "${oldValue}", new value is "${newValue}".`
+    );
   }
-  // Save it using the Chrome extension storage API.
-  chrome.storage.sync.set({'value': theValue}, function() {
-    // Notify that we saved.
-    message('Settings saved');
-  });
-}
+});
 ```
 
-If you're interested in tracking changes made to a data object, you can add a listener to its
-`onChanged` event. Whenever anything changes in storage, that event fires. Here's sample code to
-listen for saved changes:
+We can take this idea even further. In this example we have an [options page][options-page] that
+allows the user to toggle a "debug mode" in the extension (implementation not shown here). Changes
+to this setting are immediately saved to sync storage by the options page and the background script uses `storage.onChanged` to apply the setting as soon as possible.
 
 ```js
-chrome.storage.onChanged.addListener(function(changes, namespace) {
-  for (var key in changes) {
-    var storageChange = changes[key];
-    console.log('Storage key "%s" in namespace "%s" changed. ' +
-                'Old value was "%s", new value is "%s".',
-                key,
-                namespace,
-                storageChange.oldValue,
-                storageChange.newValue);
+//// options.html ////
+
+<script defer src="options.js"></script>
+<form id="optionsForm">
+  <label for="debug">
+    <input type="checkbox" name="debug" id="debug">
+    Enable debug mode
+  </label>
+</form>
+
+//// options.js ////
+
+// In-page cache of the user's options
+let options = {};
+
+// Initialize the form with the user's option settings
+chrome.storage.sync.get('options', (data) => {
+  Object.assign(options, data?.options);
+  optionsForm.debug.checked = data?.options?.debug;
+});
+
+// Immediately persist options changes
+optionsForm.debug.addEventListener('change', (event) => {
+  options.debug = event.target.checked;
+  chrome.storage.sync.set({options});
+});
+
+//// background.js ////
+
+// Watch for changes to the user's options & apply them
+chrome.storage.onChanged.addListener((changes, area) => {
+  if (area === 'sync' && changes.options) {
+    console.log('enable debug mode?', !!changes.options?.newValue?.debug);
```

Comment:
same here: `changes.options?.` is already known to be truthy so `?.` is redundant and confusing.

Reply:
I slightly disagree. I think a clearer way to write this is:

```js
  if (area === 'sync' && changes.options?.newValue) {
    const debugMode = Boolean(changes.options.newValue.debug);
    console.log('blah', debugMode);
    setDebugMode(debugMode);
  }
```

The condition in the `if` will e.g., make VSCode/tsc happy to allow the access inside the block.

Reply:
@samthor, nope, your variant won't react to `false` values.

Reply:
Er, sorry, I missed the fact that `debug` is not a separate value but a property nested inside `options`...

Reply:
Thanks, both of you. @samthor's suggestion here looks solid, I'll add that now.